src/software/ai/navigator/path_planner/simulated_theta_star_test.cpp

```diff
@@ -0,0 +1,241 @@
+         Point(front_wall_x + gate_1, -1), Point(front_wall_x + gate_2, 0.0),
+         Point(front_wall_x + gate_2, 0.5), Point(front_wall_x + gate_2, 1),
+         Point(front_wall_x + gate_3, 0.0), Point(front_wall_x + gate_3, -0.5),
+         Point(front_wall_x + gate_3, -1)}));
+
+    auto tactic = std::make_shared<MoveTactic>(false);
+    tactic->updateControlParams(destination, Angle::zero(), 0);
+    setTactic(tactic);
+    setRobotId(1);
+
+    std::set<MotionConstraint> motion_constraints;
+    motion_constraints.insert(MotionConstraint::ENEMY_ROBOTS_COLLISION);
+    setMotionConstraints(motion_constraints);
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [destination, tactic](std::shared_ptr<World> world_ptr,
+                              ValidationCoroutine::push_type& yield) {
+            Rectangle expectedFinalPosition(
+                Point(destination.x() + 0.015, destination.y() + 0.015),
+                Point(destination.x() - 0.015, destination.y() - 0.015));
+            robotInPolygon(1, expectedFinalPosition, world_ptr, yield);
+        }};
+
+    std::vector<ValidationFunction> non_terminating_validation_functions = {};
+
+    runTest(terminating_validation_functions, non_terminating_validation_functions,
+            Duration::fromSeconds(15));
+}
+
+TEST_F(SimulatedThetaStarTest, test_theta_star_oscillation)
+{
+    /*
+     * When DISTANCE_THRESHOLD (what determines if robot is close enough to destination)
+     * in transition_condition.h is lowered (eg. to 0.02), the robot oscillates back
+     * and forth and does not reach a steady state at the destination in some cases.
+     * This test is for visual purposes to check for this bug.
+     * Known destinations that oscillate with DISTANCE_THRESHOLD = 0.02:
+     *  - (-2.5,-2.5)
+     *  - (2.5,-2.5)
+     */
+    Point destination      = Point(-2.5, -2.5);
+    Point initial_position = Point(-3, 1.5);
+    setBallState(BallState(Point(0, 0), Vector(0, 0)));
+    addFriendlyRobots(
+        TestUtil::createStationaryRobotStatesWithId({Point(-3, 0), initial_position}));
+    addEnemyRobots(TestUtil::createStationaryRobotStatesWithId({Point(1, 0)}));
+
+    auto tactic = std::make_shared<MoveTactic>(false);
+    tactic->updateControlParams(destination, Angle::zero(), 0);
+    setTactic(tactic);
+    setRobotId(1);
+
+    std::set<MotionConstraint> motion_constraints;
+    motion_constraints.insert(MotionConstraint::ENEMY_ROBOTS_COLLISION);
+    setMotionConstraints(motion_constraints);
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [destination, tactic](std::shared_ptr<World> world_ptr,
+                              ValidationCoroutine::push_type& yield) {
+            while (world_ptr->getMostRecentTimestamp() < Timestamp::fromSeconds(9.5))
+            {
+                yield("Timestamp not at 9.5s");
+            }
```

Comment:
Can we check that the robot stabilizes to a single location? Might as well do the same for all the other theta star path planner tests

Reply:
Would that be a new terminating validation function similar to `robot_in_polygon_validation` which also checks that the velocity is 0 once its in the polygon? Or would it look different?

Reply:
Yeah that makes sense; we'd like to check this validation function over hundreds of ticks to make sure it stays stopped.

Reply:
Added a new validation function. Let me know if you think anything should be changed